hack/tests/ci-prepare.sh

```diff
@@ -30,6 +30,31 @@ stop_monitoring() {
     fi
 }
 
+cleanup() {
+    set +e
+
+    if [[ -n "$ARTIFACTS" ]]; then
+        exec &>"$ARTIFACTS/cleanup"
+    fi
+
+    if [[ -n "$MONITORING" ]]; then
```

Comment:
do you need this check as stop_monitoring only does something if MON_PID is not empty